libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp

```diff
@@ -42,14 +43,19 @@ void SyncingState::set_pbft_syncing(bool syncing, uint64_t current_period, std::
     set_last_sync_packet_time();
   } else {
     deep_pbft_syncing_ = false;
-    peer_ = nullptr;
+    // We can set peer to nullptr only if we are not dag_syncing_
+    if (!dag_syncing_) peer_ = nullptr;
   }
 }
 
-void SyncingState::set_dag_syncing(bool syncing, std::shared_ptr<TaraxaPeer> peer) {
+void SyncingState::set_dag_syncing(bool syncing, std::shared_ptr<TaraxaPeer> peer /*=nullptr*/) {
   assert((syncing && peer) || !syncing);
   dag_syncing_ = syncing;
-  set_peer(std::move(peer));
+  // We set a peer if peer is not null and we are syncing or
+  // we set peer to null if dag and pbft syncing are not enabled
+  if ((syncing && peer) || (!syncing && !pbft_syncing_)) {
+    set_peer(std::move(peer));
```

Comment:
I think we are basically checking the same thing in two places in set_pbft_syncing and set_dag_syncing, maybe on both places we only call set_peer and then inside of set_peer we do the check. But I am fine as is as well

Reply:
you are right, that is better